### PR TITLE
Mecha Kinetic Accelerator

### PR DESCRIPTION
**Pull request by KazeEspada**
Adds a brand new mecha kinetic accelerator. The basic model, which you can make as soon as you can make a ripley is as powerful as a hyper-KA.

If you get your tech levels high enough(which is really easy) you can make the Mecha Hyper-KA which mines a whopping 5 by 5 squares!

Also, practically obsoleted the can_attach() proc with a new fancy var called compatibleMechs. Its a list of types that makes it much easier to determine which items can be held by which mechs.

### CODE REVIEW

**Comment by Bawhoppen** [comment]
this is OP if all it takes is metal, make the basic one require plasma and glass atleast, and the hyper one should require as much stuff as the actual hyper KA

**Comment by PKPenguin321** [comment]
Not really baw, since mechs are so clunky anyways

**Comment by Bawhoppen** [comment]
there should be some sort of limit more than just having a few sheets of metal

**Comment by PKPenguin321** [comment]
There is, it's called creating and having to use a mech

**Comment by Bawhoppen** [comment]
all it takes is a ripley which is makeable from roundstart metals, just add some silver to the cost and call it a day

File: code/modules/research/designs/mechfabricator_designs.dm [hunk 1]
```diff
@@ -567,6 +567,28 @@
...
+	category = list("Exosuit Equipment")
+
+/datum/design/mech_hyper_ka
+	name = "Exosuit Weapon (LL-KA mounted hyper kinectic accelerator)"
+	desc = "Allows for the construction of LL-HA mounted hyper kinectic accelerator"
+	id = "mech_hyper_ka"
+	build_type = MECHFAB
+	req_tech = list("combat" = 4, "powerstorage" = 3)
+	build_path = /obj/item/mecha_parts/mecha_equipment/weapon/energy/ka/hyper
+	materials = list(MAT_METAL=20000)
```
Comment (GunHog) [inline code/modules/research/designs/mechfabricator_designs.dm]: Should require more minerals.

Comment (KorPhaeron) [inline code/modules/research/designs/mechfabricator_designs.dm]: Yeah. You have rendered all mining upgrades and progression irrelevant with a grand total of 10 sheets of metal.

**Comment by Aloraydrel** [comment]
YES

File: code/modules/research/designs/mechfabricator_designs.dm (hunk 1 above)
Comment (KazeEspada) [inline code/modules/research/designs/mechfabricator_designs.dm]: Its as expensive as the weapons in the exofab but i do think it needs a price increase. Suggestions please.

**Comment by Jazaen** [comment]
Well, while the idea is sound it should be at least a bit harder to get/weaker, so it's not instant win Übermining tool

File: code/modules/research/designs/mechfabricator_designs.dm (hunk 1 above)
Comment (KorPhaeron) [inline code/modules/research/designs/mechfabricator_designs.dm]: Most mech weapons do not destroy nearly a screens worth of tiles in one shot.

And yes most stuff in RD is undercosted.

**Comment by KazeEspada** [comment]
I personally think due to the clunkiness of mechs, the mech version of an item must always be much more superior to the non mech version. However, the price right now is much too cheap, and would really like suggestions to bring it to a good amount.

**Comment by KorPhaeron** [comment]
There is a difference between super and blowing up the whole fucking screen

**Comment by KazeEspada** [comment]
Its no where near the whole screen.

**Comment by TheNightingale** [comment]
The lesser version is good as it is, but for the more powerful one, 5x5 is actually pretty huge, and given that it requires little tech to get (Combat 4 and Power 3 aren't big deals at all), it really should require some more exotic minerals. Is it more powerful than a mech diamond drill? (Test it and see which one mines faster.)

**Comment by KazeEspada** [comment]
Diamond drill collects loot and can damage station walls.

**Comment by KorPhaeron** [comment]
Cant it just be the same strength as the normal KA (maybe a bit more damage or lower cooldown?) and leave the drill for mining?

**Comment by KazeEspada** [comment]
If it was the same damage, why the would i get the mech?

**Comment by KorPhaeron** [comment]
Because the mech is ash proof and has the giant drill on it

**Comment by KazeEspada** [comment]
Mechs are also slow and clunky.

**Comment by KorPhaeron** [comment]
1 week and no activity/no changes I requested, reopen it if you're willing to make the changes

**Comment by KazeEspada** [comment]
Ive been busy.

**Pull request closed without merging**